frontend/spec/features/checkout_confirm_insufficient_stock_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe "Checkout confirm page submission", type: :feature do
+  include_context 'checkout setup'
+
+  context "when the product from the order is not backorderable but has enough stock quantity" do
+    let(:user) { create(:user) }
+
+    let(:order) { Spree::TestingSupport::OrderWalkthrough.up_to(:payment) }
+    let(:order_product) { order.products.first }
+    let(:order_stock_item) { order_product.stock_items.first }
+
+    before do
+      order_stock_item.update! backorderable: false
+      order_stock_item.set_count_on_hand(1)
+      allow_any_instance_of(Spree::CheckoutController).to receive_messages(current_order: order)
+      allow_any_instance_of(Spree::CheckoutController).to receive_messages(try_spree_current_user: user)
+      allow_any_instance_of(Spree::OrdersController).to receive_messages(try_spree_current_user: user)
+    end
+
+    context 'when there are not other backorderable stock locations' do
+      context 'when the customer is on the confirm page and the availabilty drops to zero' do
+        before do
+          visit spree.checkout_state_path(:confirm)
+          order_stock_item.set_count_on_hand(0)
+        end
+
+        it 'redirects to cart page and shows an unavailable product message' do
+          click_button "Place Order"
+          expect(page).to have_content "#{order_product.name} became unavailable"
+          expect(page).to have_current_path spree.cart_path
```

Comment:
not sure how this could pass if the path in the controller was wrong, can you please double check?

Reply:
Specs pass because they don't exercise the path that was wrong. I'll fix the path and add another test for that branch of the `if`.